### PR TITLE
add masked_lm test

### LINKED ISSUE #4300: Add unit test suite for masked_lm task
## 🚀 Feature Request
Add a unit test for masked_lm task

### Motivation

I can't find any unit test for a FairseqTask that could help me understand the way it works in Fairseq. I'd like to understand how it works to implement a new one.

### Pitch

I want a unit test suite to be added for a `MaskedLMTask`. I'm willing to contribute.

### PR DESCRIPTION
**Pull request by clumsy**
# Before submitting

- [X] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [X] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/main/CONTRIBUTING.md)?
- [X] Did you make sure to update the docs?
- [X] Did you write any new necessary tests?

## What does this PR do?
Fixes #4300

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Big time!

Note:
I had to update `black` because of [this known issue](https://github.com/psf/black/issues/2964):
```
black....................................................................Failed
- hook id: black
- exit code: 1
Traceback (most recent call last):
  File "/Users/azzhipa/.cache/pre-commit/repoxt83whf2/py_env-python3.8/bin/black", line 8, in <module>
    sys.exit(patched_main())
  File "/Users/azzhipa/.cache/pre-commit/repoxt83whf2/py_env-python3.8/lib/python3.8/site-packages/black/__init__.py", line 1423, in patched_main
    patch_click()
  File "/Users/azzhipa/.cache/pre-commit/repoxt83whf2/py_env-python3.8/lib/python3.8/site-packages/black/__init__.py", line 1409, in patch_click
    from click import _unicodefun
ImportError: cannot import name '_unicodefun' from 'click' (/Users/azzhipa/.cache/pre-commit/repoxt83whf2/py_env-python3.8/lib/python3.8/site-packages/click/__init__.py)
```

### CODE REVIEW

**Comment by chauhang** [comment]
cc: @dianaml0

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ repos:
     -   id: end-of-file-fixer
 
 -   repo: https://github.com/ambv/black
-    rev: 22.1.0
+    rev: 22.3.0
```
Comment (dianaml0) [inline .pre-commit-config.yaml:20]: Could you pin the version in `.github/workflows/build.yml` as well? 

Does this require any updates to other files?

**Review by dianaml0 — CHANGES_REQUESTED** [review]
Thanks for adding the test! Looks good! Just had one comment

File: .pre-commit-config.yaml (hunk 1 above)
Comment (clumsy) [reply .pre-commit-config.yaml:20]: Sure, I've pinned the version of `black` in github `build` workflow and force-pushed the new commit. Please let me know if more changes are needed, I can't see the results of the internal Phabricator build.

Comment (dianaml0) [reply .pre-commit-config.yaml:20]: Thanks for the changes!

**Review by dianaml0 — APPROVED** [review]

**Pull request closed without merging**